src/components/Toast/Toast.tsx

```diff
@@ -1,92 +1,75 @@
-  closeButtonText?: string;
+  text: React.ReactNode;
   /**
-   * Toggles the ability to dismiss the toast via an close button in the top right of the toast
+   * The color of the Toast, based on EDS defined colors. Also determines the icon used.
+   * Note that the Icon mapping matches the style of Banners.
    */
-  dismissible?: boolean;
+  variant: Variant;
   /**
-   * Name of toast SVG icon. Available options are: toast, error, warning-2, success
+   * Callback when Toast is dismissed.
    */
-  iconName?: string;
-  /**
-   * The title attribute for the SVG icon in the toast
-   */
-  iconTitle?: string;
-  /**
-   * Stylistic variations for the toast type.
-   * - **success** - results in a green toast
-   * - **warning** - results in a yellow toast
-   * - **error** - results in a red toast
-   * - **info** - results in a blue toast
-   */
-  variant?: 'success' | 'warning' | 'error' | 'info' | 'brand';
-}
+  onDismiss?: () => void;
+};
 
 /**
- * Primary UI component for user interaction
+ * ```ts
+ * import {Toast} from "@chanzuckerberg/eds-components";
+ * ```
+ *
+ * A toast used to provide information on the state of the page, usually in response to a
+ * user action. Ex: The user updates their profile, and a toast pop-up informs them that the
+ * data was successfully saved.
  */
-export const Toast: React.FC<Props> = ({
-  iconTitle,
+export const Toast = ({
+  text,
   className,
-  children,
-  iconName,
-  dismissible = true,
   variant,
-  closeButtonText = 'Close',
-  ...other
-}) => {
-  const componentClassName = clsx(styles['toast'], className, {
-    [styles['toast--success']]: variant === 'success',
-    [styles['toast--warning']]: variant === 'warning',
-    [styles['toast--error']]: variant === 'error',
-    [styles['toast--brand']]: variant === 'brand',
-  });
-  const [dismissed, setDismissed] = useState(false);
-
-  function onDismiss(e: any) {
-    e.preventDefault();
-    setDismissed(true);
-  }
-  return dismissed ? null : (
-    <div className={componentClassName} role="alert" {...other}>
-      {iconName && (
-        <Icon
-          name={iconName}
-          title={iconTitle}
-          className={styles['toast__icon']}
-        />
-      )}
-      <div className={styles['toast__body']}>{children}</div>
-
-      {dismissible && (
-        <Button
-          className={styles['toast__close-btn']}
-          variant="bare"
-          hideText={true}
-          text={closeButtonText}
-          iconName="x"
-          iconPosition="after"
-          onClick={(e: any) => onDismiss(e)}
-        />
+  onDismiss,
+  // Allow for additional attributes such as aria roles
+  ...rest
+}: Props) => {
+  return (
+    <div
+      className={clsx(
+        className,
+        styles.toast,
+        /* TODO: uncomment and point to corresponding style classes once Notifications is fully migrated */
+        // variant === 'success' && colorStyles.colorSuccess,
+        // variant === 'alert' && colorStyles.colorAlert,
```

Comment:
I wonder if those would be good candidates for mixins.

Reply:
In any case, I'm excited to see what we'll end up doing for these common style-type things.

Reply:
🤔

Reply:
ooh yeah interesting. @bradfrost @ifrost1 @tpitre-czi for reference, we have a style file shared between `Banner` and `Toast` that contains things like:
```
// common/Notifications/Notification.module.css

.colorBrand {
  color: var(--eds-color-brand-700);
  background-color: var(--eds-color-brand-100);
  --border-light-color: var(--eds-color-brand-200);
  --border-dark-color: var(--eds-color-brand-500);
}
```

are mixins a better way to handle this type of style sharing? something like `@define-mixin notificationSuccess` in `src/design-tokens/mixins.css`?